### DISCUSSION TITLE
@exclude-match vs. @exclude (performance vs. compatibility?)

### DISCUSSION THREAD

**Discussion by e-t-l** [erosman/support#577] [Q&A]
Firemonkey Help documentation states that `@exclude-match` is better for performance than `exclude`, but that the former is _not_ supported by Greasemonkey and Tampermonkey. If I am writing a userscript intended to be shared (i.e. with users who may use different extensions/browsers besides Firemonkey on Firefox), should I use `@exclude` so that the script is more broadly compatible? (I know the performance for using one type of rule vs. the other difference is probably negligible for most userscripts, so this is partly a theoretical question).

**Comment by erosman** [comment]
FireMonkey attempts to convert `include/exclude` to `match/exclude-match`, therefore, as long as the pattern is a proper [Match Patterns](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/match_patterns), it will be fine.

I would be better to use `@match` if possible, as it is supported by all userscript managers.
